test/fuzz/transformation_changing_memory_semantics_test.cpp

```diff
@@ -0,0 +1,526 @@
+                   MakeInstructionDescriptor(150, SpvOpAtomicLoad, 0), 0, 100)
+                   .IsApplicable(context.get(), transformation_context));
+
+#ifndef NDEBUG
+
+  // Bad: Operand position does not equal 0 or 1.
+  ASSERT_DEATH(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 2, 100)
+                   .IsApplicable(context.get(), transformation_context),
+               "The instruction may not be an atomic or barrier instruction. \
+                The operands position may be not equal 0 or 1. \
+                The position may be equal to one and the expected is zero.");
+
+  // Bad: The SpvOpAtomicLoad takes one memory semantic value and the operand
+  // position passed is 1.
+  ASSERT_DEATH(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 1, 100)
+                   .IsApplicable(context.get(), transformation_context),
+               "The instruction may not be an atomic or barrier instruction. \
+                The operands position may be not equal 0 or 1. \
+                The position may be equal to one and the expected is zero.");
+#endif
+  // Bad: The new Id instruction does not exist.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 0, 101)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The new Id instruction exists, but not OpConstant.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 0, 13)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad transformation:
+  // OpAtomicStore Release | UniformMemory
+  // to:
+  // OpAtomicStore AcquireRelease | UniformMemory
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(24, SpvOpAtomicStore, 0), 0, 100)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad transformation:
+  // OpAtomicExchange AcquireRelease | UniformMemory
+  // to:
+  // OpAtomicExchange None           | UniformMemory
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(23, SpvOpAtomicExchange, 0), 0, 97)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Value (one) is pointing to the unequal position operand.
+  auto memory_semantics_unequal_position = 1;
+  // Bad: The SpvOpAtomicCompareExchange(read-modify-write) takes two memory
+  // semantics operands. The second operand cannot be AcquireRelease value.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(25, SpvOpAtomicCompareExchange, 0),
+                   memory_semantics_unequal_position, 100)
+                   .IsApplicable(context.get(), transformation_context));
```

Comment:
Very nice comments!

However, I would add the information about the two operands being called "Equal" and "Unequal" here, just because you have introduced a variable called `memory_semantics_unequal_position`. So you are using the "unequal" term here, without explaining it. Or, you could rename `memory_semantics_unequal_position` to `memory_semantics_operand_second_position`.

Reply:
Thanks for the clarification.

Reply:
Thanks for the clarification.
Done.